pkgs/build-support/setup-hooks/audit-tmpdir.sh

```diff
@@ -15,25 +15,50 @@ auditTmpdir() {
 
     echo "checking for references to $TMPDIR/ in $dir..."
 
-    local i
-    find "$dir" -type f -print0 | while IFS= read -r -d $'\0' i; do
-        if [[ "$i" =~ .build-id ]]; then continue; fi
+    local tmpdir="$(mktemp -d)"
+    local elf_fifo="$tmpdir/elf"
+    local script_fifo="$tmpdir/script"
+    mkfifo "$elf_fifo" "$script_fifo"
 
-        if isELF "$i"; then
-            if { printf :; patchelf --print-rpath "$i"; } | grep -q -F ":$TMPDIR/"; then
-                echo "RPATH of binary $i contains a forbidden reference to $TMPDIR/"
+    # Classifier: identify ELF and script files
+    (
+        find "$dir" -type f -not -path '*/.build-id/*' -print0 \
+        | while IFS= read -r -d $'\0' file; do
+            if isELF "$file"; then
+                printf '%s\0' "$file" >&3
+            elif isScript "$file"; then
+                if [ -e "$(dirname "$file")/.$(basename "$file")-wrapped" ]; then
+                    printf '%s\0' "$file" >&4
+                fi
+            fi
+        done
+        exec 3>&- 4>&-
+    ) 3> "$elf_fifo" 4> "$script_fifo" &
```

Comment:
We potentially don't need the fifo here. In the subshells below we could use:

```
exec 3>&0
```

before running xargs to redirect fd 3 to stdin.

Reply:
I think the fifos are needed, because I cannot simply pipe from find into two different xargs processes

Reply:
Or in other words. I don't know how I would forward fd 3 and fd 4 from inside the sub-shell to 2x xargs without fifo.

Reply:
Right. It looks like bash isn't capable of that. In other languages one could open up pipes like that.